src/librustc_borrowck/borrowck/mir/dataflow/impls.rs

```diff
@@ -442,6 +442,7 @@ impl<'a, 'tcx> BitDenotation for MovingOutStatements<'a, 'tcx> {
         }
         let bits_per_block = self.bits_per_block(ctxt);
         match stmt.kind {
+            repr::StatementKind::SetDiscriminant { ref lvalue, .. } |
```

Comment:
Nit: I'd probably prefer to `span_bug!`, since we don't expect to see this here, and it's not clear to me what we ought to do if ever *do* try to run borrowck when set-discriminant operations are running around.